src/dataset/create_dataset.py

```diff
@@ -19,240 +20,141 @@
-                    
-                    # Get title, author, date, URL, tags, and text
-                    title, author, date_published, url, tags, text = self.extract_info_from_article(entry)
-                    
-                    # If there's less than 2 of 'title', 'author' and 'url', ignore this text
-                    if (((title or '').strip() == '') + ((author or '').strip() == '') + ((url or '').strip() == '')) > 1:
-                        print(f'{entry["source"]}')
-                        continue
-                    
-                    #if the text is too short, ignore this text
-                    if len(text) < 500:
-                        continue
-
-                    #we're keeping the text so we inc the aticle count
-                    self.articles_count[entry['source']] += 1
-                    self.total_articles_count += 1
-                    
-                    # Get signature
-                    signature = ""
-                    if title: signature += f"Title: {title}, "
-                    else: signature += f"Title: None, "
-                    if author: signature += f"Author: {author}"
-                    else: signature += f"Author: None"
-                    # if date_published: signature += f"Date published: {date_published}, "
-                    # if url: signature += f"URL: {url}, "
-                    # if tags: signature += f"Tags: {tags}, "  # Temporary decision to not include tags in the signature
-                    # if signature: signature = signature[:-2]
-                    signature = signature.replace("\n", " ")
-                    
-                    # Add info to metadata and embedding strings
-                    self.metadata.append((title, author, date_published, url, tags))
-                    blocks = text_splitter.split(text, signature)
-                    self.embedding_strings.extend(blocks)
-                    self.embeddings_metadata_index.extend([self.total_articles_count-1] * len(blocks))
-                    
-                    # Update counts
-                    self.total_char_count += len(text)
-                    self.total_word_count += len(text.split())
-                    self.total_sentence_count += len(split_into_sentences(text))
-                    self.total_block_count += len(blocks)
-                    
-                except MissingDataException as e:
-                    if str(e) not in error_count_dict:
-                        error_count_dict[str(e)] = 0
-                    error_count_dict[str(e)] += 1
+
+        # Load the dataset. streaming allows you to load one entry at a time, 
+        # so entries can be processed before the entire dataset has been saved.
+        # iterable_data = load_dataset('StampyAI/alignment-research-dataset', 'aisafety.info', split='train', streaming=True)
+
+        iterable_data = load_dataset('StampyAI/alignment-research-dataset', 'all', split='train', streaming=True)
```

Comment:
how about extracting this to a method (or even property), which would be something like:
```
def iterable_data(self):
    if not self.custom_sources:
        return tqdm(load_dataset('StampyAI/alignment-research-dataset', 'all', split='train', streaming=True))
    
    data = (entry for source in self.custom_sources for entry in load_dataset('StampyAI/alignment-research-dataset', source, split='train', streaming=True))
    return tqdm(data)
```

Reply:
my brother, who is currently creating a new feature from my branch, did something similar to this, makes sense